src/python/tests/gff3/test_restructure.py

```diff
@@ -0,0 +1,314 @@
+# See the NOTICE file distributed with this work for additional information
+# regarding copyright ownership.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Unit testing of `ensembl.io.genomio.gff3.restructure` module."""
+
+from contextlib import nullcontext as does_not_raise
+from typing import Any, ContextManager, Dict, List, Union
+
+from Bio.SeqFeature import SeqFeature, SimpleLocation
+import pytest
+from pytest import param, raises
+
+from ensembl.io.genomio.gff3.exceptions import GFFParserError
+from ensembl.io.genomio.gff3.restructure import (
+    restructure_gene,
+    add_transcript_to_naked_gene,
+    move_only_cdss_to_new_mrna,
+    move_only_exons_to_new_mrna,
+    move_cds_to_existing_mrna,
+    remove_extra_exons,
+)
+
+
+class FeatGenerator:
+    """Generates features and structures for testing."""
+
+    start = 1
+    end = 1000
+    strand = -1
+    region = "LOREM"
+    source = "Foo"
+
+    def make(self, ftype: str, number: int = 1) -> List[SeqFeature]:
+        """Create a defined number of features of a given type."""
+        feats = []
+        for _ in range(0, number):
+            loc = SimpleLocation(self.start, self.end, self.strand)
+            feat = SeqFeature(loc, type=ftype)
+            feat.qualifiers["source"] = self.source
+            feat.sub_features = []
+            feats.append(feat)
```

Comment:
I was about to suggest a list comprehension approach, but this would just provide the same object over and over, so editing one edits everything. Nice implementation!